### PR TITLE
Add optional JSON output

### PR DESCRIPTION
**Pull request by wileykestner**
- Allow users to select a custom reporter that prints JSON instead of `sed` commands

- Opt-in by adding `"reporter": "json"` to the configuration file

### CODE REVIEW

File: Sources/unused-imports/main.swift [hunk 1]
```diff
@@ -51,6 +54,16 @@ private struct Configuration: Decodable {
         } else {
             self.ignoredModuleRegex = nil
         }
+
+        if let string = try values.decodeIfPresent(String.self, forKey: .reporter) {
+            if string == "json" {
+                self.reporter = JSONReporter()
+            } else {
+                self.reporter = nil
```
Comment (keith) [inline Sources/unused-imports/main.swift:62]: can we make this case error for unhandled values?

File: Sources/unused-imports/main.swift [hunk 2]
```diff
@@ -51,6 +54,16 @@ private struct Configuration: Decodable {
...
+            }
+        } else {
+            self.reporter = nil
```
Comment (keith) [inline Sources/unused-imports/main.swift:65]:
```suggestion
            self.reporter = SedCommandReporter()
```

seems fine to default this here so you can eliminate the optionality

**Review by keith — COMMENTED** [review]
looks good! can you ensure newlines at the end of the files

File: Sources/unused-imports/main.swift (hunk 1 above)
Comment (wileykestner) [reply Sources/unused-imports/main.swift:62]: Absolutely.  Great suggestion.

File: Sources/unused-imports/main.swift (hunk 2 above)
Comment (wileykestner) [reply Sources/unused-imports/main.swift:65]: Absolutely.  100%.

File: Sources/unused-imports/main.swift (hunk 1 above)
Comment (wileykestner) [reply Sources/unused-imports/main.swift]: Updated.

File: Sources/unused-imports/main.swift (hunk 2 above)
Comment (wileykestner) [reply Sources/unused-imports/main.swift]: Updated.

**Review by keith — APPROVED** [review]

**Comment by keith** [comment]
thanks!

**Pull request merged**